src/wfcopy.c

```diff
@@ -2339,8 +2339,9 @@ WFMoveCopyDriverThread(LPVOID lpParameter)
          // deal with case where directory is implicit in source
          // move/copy: *.* -> c:\windows, c:\windows -> c:\temp
          // or foo.bar -> c:\temp
+         // do this not for jucntion or symlinks
 
-         if (!IsWild(pCopyInfo->pTo) && (ManySource || IsDirectory(pCopyInfo->pTo))) {
+         if (!IsWild(pCopyInfo->pTo) && (ManySource || IsDirectory(pCopyInfo->pTo)) && pCopyInfo->dwFunc != FUNC_LINK && pCopyInfo->dwFunc != FUNC_JUNC) {
             AddBackslash(pCopyInfo->pTo);

```

Comment:
Based on FUNC_JUNC and FUNC_HARD both go to WFJunction(), why do this for one and not the other?

Reply:
The above check is either for FUNC_JUNC or FUNC_LINK
We need this here because the 'drap/drop coding' provides pCopyInfo->pTo already with appended *.*, but the 'menu coding' does not
So we must filter it and append *.* so that the later hydraulics work.

Reply:
The above check is either for FUNC_JUNC or FUNC_LINK, which go different ways in the end.

- The 'menu code' already provides the fullly assembled destination for the symlink/junction. One must not append a \*.\*
- The 'drag/drop code' provides the path where the destination should go in and has \*.\* appended (which is later stripped off)

So this code is here to have WFMoveCopyDriverThread() bullet proof for input like that.